gst/nnstreamer/tensor_query/tensor_query_serversink.c

```diff
@@ -134,6 +134,79 @@ gst_tensor_query_serversink_finalize (GObject * object)
   G_OBJECT_CLASS (parent_class)->finalize (object);
 }
 
+/**
+ * @brief start processing of query_serversink
+ */
+static void
+_gst_tensor_query_serversink_start (GstTensorQueryServerSink * sink)
+{
+  g_autofree gchar *id_str = NULL;
+
+  id_str = g_strdup_printf ("%u", sink->sink_id);
+  sink->server_h = gst_tensor_query_server_add_data (id_str);
+
+  gst_tensor_query_server_set_configured (sink->server_h);
+}
+
+/**
+ * @brief start processing of query_serversink
+ */
+static void
+_gst_tensor_query_serversink_playing (GstTensorQueryServerSink * sink)
+{
+  g_autofree gchar *id_str = NULL;
+
+  id_str = g_strdup_printf ("%u", sink->sink_id);
+  sink->edge_h =
+      gst_tensor_query_server_get_edge_handle (id_str, sink->connect_type);
+}
+
+/**
+ * @brief Change state of query server sink.
+ */
+static GstStateChangeReturn
+gst_tensor_query_serversink_change_state (GstElement * element,
+    GstStateChange transition)
+{
+  GstTensorQueryServerSink *sink = GST_TENSOR_QUERY_SERVERSINK (element);
+  GstStateChangeReturn ret = GST_STATE_CHANGE_SUCCESS;
+
+  switch (transition) {
+    case GST_STATE_CHANGE_PAUSED_TO_PLAYING:
+      _gst_tensor_query_serversink_playing (sink);
+      if (!sink->edge_h) {
+        nns_loge ("Failed to change state from PAUSED to PLAYING.");
+        return GST_STATE_CHANGE_FAILURE;
+      }
+      break;
+    case GST_STATE_CHANGE_READY_TO_PAUSED:
+      _gst_tensor_query_serversink_start (sink);
+      if (!sink->server_h) {
+        nns_loge ("Failed to change state from READY to PAUSED.");
+        return GST_STATE_CHANGE_FAILURE;
+      }
+    default:
+      break;
+  }
+
+  ret = GST_ELEMENT_CLASS (parent_class)->change_state (element, transition);
+  if (ret != GST_STATE_CHANGE_SUCCESS) {
```

Comment:
```suggestion
  if (ret == GST_STATE_CHANGE_FAILURE) {
```

state can be async or no preroll :)

Reply:
Right. For that case, it is not a failure, but it will be judged as a failure. Thanks. Updated.